src/agent/coverage/src/cobertura.rs

```diff
@@ -18,12 +19,31 @@ pub fn cobertura(source_coverage: SourceCoverage) -> Result<String, Error> {
         .context("system time before unix epoch")?
         .as_secs();
 
+    // compute line rate, lines-covered, lines-valid by summing total lines and summing lines that have at least 1 hit, will do this for overall, and for each file (package and class in this case)
+    let mut total_valid_lines = 0;
```

Comment:
This block, and the one below, are doing something very similar. Can you factor this duplicate logic out into a shared function?

Reply:
Seperated out into a couple different functions, wasn't able to completely unduplicate.